src/braft/log.cpp

```diff
@@ -94,6 +94,7 @@ int Segment::create() {
         return -1;
     }
 
+    int old_fd = _fd;
```

Comment:
The pr is about the bug is in Segment::close, why all the modification is in create()?

Reply:
The issue is:
Segment does not close the fd when it open a new log.
My first commit close this fd in Segment::close method, but that will break the ut.

Reply:
Or I need close this pr and create a new one?

Reply:
`create()` is just another kind of `initialization`. It's not supposed that fd is valid when this function is invoked. Add a `CHECK` if you concern about this stuff.

Reply:
This assignment does not assume fd is invalid, the check you request is under below:
old_fd >= 0 ? ::close(old_fd) : 0

Only need to close old_fd when it's valid.